core/src/main/java/org/commonjava/maven/ext/core/groovy/BaseScript.java

```diff
@@ -154,20 +154,28 @@ public Translator getRESTAPI() throws ManipulationException
      * @param project Current project
      * @param stage the current InvocationStage of the groovy script
      */
-    public void setValues(PomIO pomIO, FileIO fileIO, ModelIO modelIO, ManipulationSession session, List<Project> projects, Project project,
-                          InvocationStage stage)
+    public void setValues( PomIO pomIO, FileIO fileIO, ModelIO modelIO, ManipulationSession session,
+                           List<Project> projects, Project project, InvocationStage stage )
+            throws ManipulationException
     {
         this.fileIO = fileIO;
         this.pomIO = pomIO;
         this.modelIO = modelIO;
-        this.sessionHandler = session;
+        this.session = session;
         this.projects = projects;
+
+        if (project != null) {
+            this.gav = project.getKey();
+            this.basedir = project.getPom().getParentFile();
+        } else {
+            this.basedir = session.getPom().getParentFile();
```

Comment:
`baseDir` should the same from either of these so this seems redundant (if its not the same I think we have a bug!)

Aside : formatting?

Reply:
Agreed.

Reply:
I think sometimes session pom is `null` though so that can't be called unconditionally. It works since in one one case we don't have project we do have the session.

Reply:
I think that needs to be narrowed down? - i.e. when it might be null?

Reply:
It is actually `session.getPom()`. The `TestUtil.createSession()` method does not set a pom. In most cases, there is a `File rootproject` to use for a pom, but in some cases there is none.

Reply:
If its test code creating not fully populated object we should fix that rather than adding if-blocks here in my opinion